### ISSUE TITLE
Simplify and optimize retrieval of the ActivityGraph

### ISSUE THREAD

**Issue by Daniel-Miller** [InSite/Lotus#32]
As per our conversation about this a few weeks ago, I have started to transform our entities from JSON documents (stored in the Document table) into 'native' EF entities.

This seems like this has been a good idea. It has eliminated a lot of duplication, and it makes debugging much easier.  

Some of the code seems (to me) simpler as well. For example, notice my approach with the method ActivityRepository.SelectPlansAsTrees. Activities in Lotus do not change, so they can be 'cached' as static member variables, and we can create tree data structures dynamically.

Can we take this same approach with the ActivityGraph data structure?

In other words, can we delete the Graph document (DELETE FROM dbo.Document WHERE EntityType = 'Graph') and write a method in the Activity Repository class that returns all activities in an ActivityGraph object?

**Comment by Daniel-Miller** [comment]
I have copied a backup of my local Lotus database to the shared Dropbox folder.

**Comment by Oleg-Terzi** [comment]
Done.

I moved all methods/fields that are work with cached data to the new static class ActivityStorage and I left the update methods in ActivityRepository class. I made these changes by following reasons:

1. I think it will be better if a cached data will be initialized in static constructor (before *any* method will be called).
2. I think it would be better when these two entities are separated. In this case we (as programmers) can easily discern when we are working with cached data and when we are working directly with database.

I hope these changes are OK.

Note: ActivityTree and ActivityTreeNode classes are not using in the Lotus project anymore but Iris project is using them and I not sure that we could remove these classes.

**Comment by Daniel-Miller** [comment]
This is an excellent strategy -- good decision here. I renamed ActivityStorage to ActivityCache, so the purpose of the new class is even more clear (i.e. static cache).  I have also removed the ActivityTree and ActivityTreeNode classes from the project -- but I have not deleted them from the repository, just in case we need them for future reference.
